src/cpp/core/include/core/json/Json.hpp

```diff
@@ -84,6 +131,46 @@ bool fillVectorString(const Array& array, std::vector<std::string>* pVector);
 bool fillVectorInt(const Array& array, std::vector<int>* pVector);
 bool fillMap(const Object& array, std::map< std::string, std::vector<std::string> >* pMap);
 
+namespace detail {
+
+inline int intField(const Object& object,
+                    const std::string& name,
+                    int ifNotFound,
+                    const ErrorLocation& location)
+{
+   if (object.count(name) == 0)
+   {
+      log::logWarningMessage("No field named '" + name + "'", location);
+      return ifNotFound;
+   }
+   
+   return const_cast<Object&>(object)[name].get_int();
```

Comment:
If this isn't an int (e.g. it's a string) what happens? My impression has been that it will throw an exception which will end up in our calling R_Suicide. I *think* you need an extra type check branch here that logs the incorrect type as a warning. See the implementation of json::readObject for what we do there.